addons/website_sale_slides/tests/test_course_purchase_flow.py

```diff
@@ -45,3 +45,42 @@ def test_course_purchase_flow(self):
         # Step 3: check that the customer is now a member of both channel
         self.assertIn(self.customer, self.channel.partner_ids)
         self.assertIn(self.customer, self.channel_2.partner_ids)
+
+    def test_course_purchase_flow_with_salesman(self):
+        course_product = self.env['product.product'].create({
+            'name': "Course Product",
+            'standard_price': 100,
+            'list_price': 150,
+            'type': 'service',
+            'invoice_policy': 'order',
+            'is_published': True,
+        })
+
+        salesman = self.env['res.users'].create({
+            'name': 'salesman',
+            'login': 'salesman',
+            'email': 'salesman007@example.com',
+            'groups_id': [(6, 0, self.env.ref('sales_team.group_sale_salesman').ids)],
+        })
+
+        self.channel.write({
+            'enroll': 'payment',
+            'product_id': course_product.id,
+            'user_id': salesman.id,
+        })
+
+        sale_order = self.env['sale.order'].with_user(salesman).create({
```

Comment:
Seems like a lot of copy/pasta from the test above.

I would override "setUpClass" instead and create a cls.user_salesman and then simply add the "with_user(self.user_salesman)" to the sale order creation in the above test.